### PR TITLE
fix: skip CAddTable fusion for some special case

### PR DESCRIPTION
**Pull request by benjamim93**
## What changes were proposed in this pull request?

A problem was detected while doing MobileNet V2 inference. MobileNet V2 contains some node connection structure like below (A->B indicates the change from input channel number A to output channel number B)
                     conv1(96 -> 96)
                               /\
                             /    \
        conv2(96 -> 24)   conv3(96 -> 24)
                          /\        |
                        /    \      |
    conv4(24 ->144) CAddTable

The original method would turn forward execution into some graph like:
                     conv1(96 -> 96)
                               /\
                             /    \
                    Identity    conv3(96 -> 24)
                          /\        |
                        /    \      |
    conv4(24 ->144) conv2(96 -> 24)

resulting in a mismatch between the input channel number and output channel number for conv4. 

(Please fill in changes proposed in this patch)

Skip fusion if the previous node of CAddTable has more than one next nodes.

## How was this patch tested?

Tested MobileNet v2. The inference works.

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/XXX

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/Fusion.scala [hunk 1]
```diff
@@ -98,7 +98,10 @@ private[mkldnn] object Fusion {
 
   private def findPrevious(node: Node[AbstractModule[Activity, Activity, Float]])
   : Node[AbstractModule[Activity, Activity, Float]] = {
-    if (node.element.isInstanceOf[Identity] && node.prevNodes.length == 1) {
+    if (node.element.isInstanceOf[Identity]
+      && node.nextNodes.length == 1
+      && node.prevNodes.length == 1
+      && node.prevNodes(0).nextNodes.length == 1) {
```
Comment (i8run) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/Fusion.scala]: Why should we add line 104 here ?

Comment (benjamim93) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/Fusion.scala]: This line is for cases like below (mentioned by @zhangxiaoli73 ):
```
                        conv1(96 -> 96)
                               |
                               |
                        conv2(96 -> 24)
                              /\        
                            /    \      
            conv3(24 ->144)    Identity    conv4(96 -> 24)
                                        \    /
                                          \/
                                        CAddTable
```

Comment (benjamim93) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/Fusion.scala]: I am not sure if there are any possible corner cases not taken into account yet.

Comment (i8run) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/Fusion.scala]: It's OK. But we should add at least two unit tests here for both cases.

Comment (benjamim93) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/Fusion.scala]: Added tests and added a condition for the case shown in the graph at the very beginning. I mean the convs without following relu, bn, etc.
